app/src/main/java/com/example/admin/learningandroid/layouts/ModelLoginActivity.java

```diff
@@ -0,0 +1,108 @@
+    public String name;
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.model_login_page);
+        etName = findViewById(R.id.et_name);
+        etRegistrationNumber = findViewById(R.id.et_registration);
+        btnSignUp = findViewById(R.id.btn_sign_up);
+        btnLogin = findViewById(R.id.btn_login);
+        
+        btnSignUp.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                //To make only one time sign up
+                SharedPreferences firstRetrieve = getSharedPreferences("SAVE_DATA", MODE_PRIVATE);
+                String savedName = firstRetrieve.getString("NAME", "");
+                String savedRegNumber = firstRetrieve.getString("REGISTRATION_NUMBER", "");
+
+                mName = etName.getText().toString();
+                mRegistrationNumber = etRegistrationNumber.getText().toString();
+
+                int mCount = mName.length();
+                //condition for various credential check
+                if (mCount <= 3) {
+                    Toast.makeText(ModelLoginPage.this, "Enter Valid Name", Toast.LENGTH_SHORT).show();
+                } else {
+                    if (mRegistrationNumber.length() < 1) {
+                        Toast.makeText(ModelLoginPage.this, "enter valid register number", Toast.LENGTH_SHORT).show();
+                    } else {
+                        Toast.makeText(ModelLoginPage.this, "Successfully saved", Toast.LENGTH_SHORT).show();
+                        if (savedName.equals("") && savedRegNumber.equals("")) {
+
+                            //Loading data in shared preference
+                            SharedPreferences saveData = getSharedPreferences("SAVE_DATA", MODE_PRIVATE);
+                            SharedPreferences.Editor addData = saveData.edit();
+                            addData.putString("NAME", mName);
+                            addData.putString("REGISTRATION_NUMBER", mRegistrationNumber);
+                            addData.apply();
+                        } else {
+                            Toast.makeText(ModelLoginPage.this, "Already signed up it seems", Toast.LENGTH_SHORT).show();
+                        }
+                    }
+                }
+            }
+
+
+        });
+
+        btnLogin.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+
+                //Retrieving shared preference data
+                SharedPreferences preference = getSharedPreferences("SAVE_DATA", MODE_PRIVATE);
+                String name = preference.getString("NAME", "");
+                String registrationNumber = preference.getString("REGISTRATION_NUMBER", "");
+
+             //making sure that the user do not logs in with empty data
+                if (name!="" && registrationNumber!="") {
```

Comment:
Try simplifying this code as well can be better structured for simplicity and readability. If Else is very convoluted